src/Entity/ItemId.php

```diff
@@ -29,12 +29,16 @@ public function __construct( $idSerialization ) {
 
 	private function assertValidIdFormat( $idSerialization ) {
 		if ( !is_string( $idSerialization ) ) {
-			throw new InvalidArgumentException( '$idSerialization must be a string; got ' . gettype( $idSerialization ) );
+			throw new InvalidArgumentException( '$idSerialization must be a string' );
 		}
 
 		if ( !preg_match( self::PATTERN, $idSerialization ) ) {
 			throw new InvalidArgumentException( '$idSerialization must match ' . self::PATTERN );
 		}
+
+		if ( strlen( $idSerialization ) > 10 && substr( $idSerialization, 1 ) > 2147483647 ) {
+			throw new InvalidArgumentException( '$idSerialization can not exceed 2147483647' );
```

Comment:
Can we have a constant for the maximum id, instead of repeating the literal everywhere?